modules/ui/org.eclipse.fx.ui.preferences/src/main/java/org/eclipse/fx/ui/preferences/PreferenceUI.java

```diff
@@ -93,11 +93,13 @@
 
 	private final PreferencePageFactory factory;
 
-	protected String currentFilter = "";
+	protected static String currentFilter = "";
 
+	private Comparator<PreferencePageProvider> comparator;
+	private Predicate<PreferencePageProvider> filter;
 	private final ObservableList<PreferencePageProvider> providers = FXCollections.observableArrayList();
-	private final SortedList<PreferencePageProvider> sortedProviders = providers.sorted(getComparator());
-	private final FilteredList<PreferencePageProvider> filteredProviders = sortedProviders.filtered(getFilter());
+	private final SortedList<PreferencePageProvider> sortedProviders = providers.sorted(comparator);
+	private final FilteredList<PreferencePageProvider> filteredProviders = sortedProviders.filtered(filter);
```

Comment:
I would initialize this field in the constructor

Reply:
Moved to constructor